### ISSUE TITLE
Heroic not running gptk app under a wineprefix, only the default wineprefix (~/.wine)

### ISSUE THREAD

**Issue by blackxfiied** [Heroic-Games-Launcher/HeroicGamesLauncher#3026]
### Describe the bug

Even when you specify a wine prefix in heroic, only the default wine prefix is used when launching games with game porting toolkit set as the wine version.

### Add logs

```text
(15:03:22) INFO:    [Legendary]:        Running command: XDG_CONFIG_HOME="/Users/blackxfiied/Library/Application Support/heroic/legendaryConfig" /opt/homebrew/bin/legendary --version
(15:03:22) INFO:    [Legendary]:        Legendary location: /opt/homebrew/bin/legendary
(15:03:22) INFO:    [Gog]:              GOGDL location: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/gogdl
(15:03:22) INFO:    [Connection]:       Connectivity: check-online
(15:03:22) INFO:    [Connection]:       Pinging external endpoints
(15:03:22) INFO:    [Backend]:          DRM module staus {
  "oimompecagnajdejgnnjijobebaeigek": {
    "name": "Widevine Content Decryption Module",
    "status": "new",
    "version": "4.10.2557.0"
  }
}
(15:03:22) WARNING: [Backend]:          Protocol already registered.
(15:03:22) INFO:    [Gog]:              Running command: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/gogdl --auth-config-path "/Users/blackxfiied/Library/Application Support/heroic/gog_store/auth.json" --version
(15:03:23) INFO:    [Backend]:          Loading Screen Ready
(15:03:23) INFO:    [Frontend]:         Refreshing undefined Library
(15:03:23) WARNING: [Backend]:          refresh not implemented on Sideload Library Manager
(15:03:23) INFO:    [Legendary]:        Refreshing library...
(15:03:23) INFO:    [Legendary]:        Refreshing Epic Games...
(15:03:23) INFO:    [Legendary]:        Game list updated, got 11 games & DLCs
(15:03:23) INFO:    [Backend]:          Frontend Ready
(15:03:23) WARNING: [Backend]:          listUpdateableGames not implemented on Sideload Library Manager
(15:03:23) INFO:    [Backend]:          Checking for new Heroic Updates
(15:03:23) INFO:    [Connection]:       Connectivity: online
(15:03:23) ERROR:   [Gog]:              Unable to syncQueued playtime, userData not present
(15:03:24) INFO:    [Legendary]:        Running command: XDG_CONFIG_HOME="/Users/blackxfiied/Library/Application Support/heroic/legendaryConfig" /opt/homebrew/bin/legendary list --third-party
(15:03:24) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(15:03:25) INFO:    [GlobalConfig]:     Searching for Game Porting Toolkit Wine
(15:03:26) INFO:    [GlobalConfig]:     Found Game Porting Toolkit Wine at /usr/local/Cellar/game-porting-toolkit/1.0.4/bin/wine64
(15:03:26) INFO:    [Winetricks]:       Downloading Winetricks
(15:03:27) WARNING: [Backend]:          Could not determine System Info
(15:03:28) INFO:    [Backend]:          Starting the Download Queue
(15:03:29) INFO:    [GlobalConfig]:     Searching for Game Porting Toolkit Wine
(15:03:29) INFO:    [GlobalConfig]:     Found Game Porting Toolkit Wine at /usr/local/Cellar/game-porting-toolkit/1.0.4/bin/wine64
(15:03:33) INFO:    [Nile]:             Running command: XDG_CONFIG_HOME="/Users/blackxfiied/Library/Application Support/heroic/nile_config" /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/nile --version
(15:06:31) INFO:    [Legendary]:        Running command: XDG_CONFIG_HOME="/Users/blackxfiied/Library/Application Support/heroic/legendaryConfig" /opt/homebrew/bin/legendary --version
(15:06:31) INFO:    [Gog]:              Running command: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/gogdl --auth-config-path "/Users/blackxfiied/Library/Application Support/heroic/gog_store/auth.json" --version
(15:06:31) INFO:    [Nile]:             Running command: XDG_CONFIG_HOME="/Users/blackxfiied/Library/Application Support/heroic/nile_config" /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/nile --version
(15:06:37) INFO:    [Backend]:          Launching Genshin Impact (41869934302e4b8cafac2d3c0e7c293d)
(15:06:37) INFO:    [Backend]:          Preventing display from sleep
(15:06:37) INFO:    [Legendary]:        Running command: XDG_CONFIG_HOME="/Users/blackxfiied/Library/Application Support/heroic/legendaryConfig" /opt/homebrew/bin/legendary --version
(15:06:37) INFO:    [Backend]:          Checking if wine version exists: GPTK Wine (DX11/DX12 Only) - wine-7.7
(15:06:37) INFO:    [Backend]:          Preventing machine to sleep
(15:06:37) INFO:    [Backend]:          Stopping Power Saver Blocker
(15:06:37) INFO:    [Gog]:              Running command: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/gogdl --auth-config-path "/Users/blackxfiied/Library/Application Support/heroic/gog_store/auth.json" --version
(15:06:37) INFO:    [Backend]:          Started Discord Rich Presence
(15:06:37) INFO:    [Backend]:          Checking if wine version exists: GPTK Wine (DX11/DX12 Only) - wine-7.7
(15:06:37) INFO:    [Backend]:          Checking if wine version exists: GPTK Wine (DX11/DX12 Only) - wine-7.7
(15:06:37) INFO:    [Backend]:          Checking if wine version exists: GPTK Wine (DX11/DX12 Only) - wine-7.7
(15:06:37) DEBUG:   [Backend]:          Running Wine command: wineboot --init
(15:06:39) INFO:    [Nile]:             Running command: XDG_CONFIG_HOME="/Users/blackxfiied/Library/Application Support/heroic/nile_config" /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/nile --version
(15:06:40) INFO:    [GlobalConfig]:     Searching for Game Porting Toolkit Wine
(15:06:40) INFO:    [GlobalConfig]:     Found Game Porting Toolkit Wine at /usr/local/Cellar/game-porting-toolkit/1.0.4/bin/wine64
(15:06:42) WARNING: [Backend]:          Could not determine System Info
(15:06:42) INFO:    [Backend]:          Launching Steam (thFgSR5VBurH1zF8QAnJ9u)
(15:06:42) INFO:    [Backend]:          Preventing display from sleep
(15:06:42) INFO:    [Legendary]:        Running command: XDG_CONFIG_HOME="/Users/blackxfiied/Library/Application Support/heroic/legendaryConfig" /opt/homebrew/bin/legendary --version
(15:06:42) INFO:    [Backend]:          Checking if wine version exists: GPTK Wine (DX11/DX12 Only) - wine-7.7
(15:06:42) INFO:    [Backend]:          Stopping Power Saver Blocker
(15:06:42) INFO:    [Gog]:              Running command: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/gogdl --auth-config-path "/Users/blackxfiied/Library/Application Support/heroic/gog_store/auth.json" --version
(15:06:42) INFO:    [Backend]:          Started Discord Rich Presence
(15:06:42) INFO:    [Backend]:          launching non-native sideloaded: /Volumes/dingus/wineapps/apps/Steam/steam.exe}
(15:06:42) INFO:    [Backend]:          Checking if wine version exists: GPTK Wine (DX11/DX12 Only) - wine-7.7
(15:06:42) DEBUG:   [Backend]:          Running Wine command: /Volumes/dingus/wineapps/apps/Steam/steam.exe 
(15:06:42) DEBUG:   [Backend]:          Logging to file "/Users/blackxfiied/Library/Application Support/heroic/GamesConfig/thFgSR5VBurH1zF8QAnJ9u-lastPlay.log"
(15:06:42) INFO:    [Legendary]:        Launching Genshin Impact: MTL_HUD_ENABLED=1 LD_PRELOAD= WINE_FULLSCREEN_FSR=0 WINEESYNC=1 WINEFSYNC=1 XDG_CONFIG_HOME="/Users/blackxfiied/Library/Application Support/heroic/legendaryConfig" /opt/homebrew/bin/legendary launch 41869934302e4b8cafac2d3c0e7c293d --language en --wine /usr/local/Cellar/game-porting-toolkit/1.0.4/bin/wine64
(15:06:43) INFO:    [GlobalConfig]:     Searching for Game Porting Toolkit Wine
(15:06:43) INFO:    [GlobalConfig]:     Found Game Porting Toolkit Wine at /usr/local/Cellar/game-porting-toolkit/1.0.4/bin/wine64
(15:06:47) WARNING: [Backend]:          Could not determine System Info
(15:06:53) INFO:    [Nile]:             Running command: XDG_CONFIG_HOME="/Users/blackxfiied/Library/Application Support/heroic/nile_config" /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/nile --version
(15:07:05) INFO:    [Backend]:          Stopped Discord Rich Presence
(15:07:05) INFO:    [Backend]:          Stopping Display Power Saver Blocker
(15:07:05) INFO:    [Frontend]:         Refreshing legendary Library
(15:07:05) INFO:    [Legendary]:        Refreshing library...
(15:07:05) INFO:    [Legendary]:        Refreshing Epic Games...
(15:07:06) INFO:    [Legendary]:        Game list updated, got 11 games & DLCs
(15:07:06) INFO:    [Legendary]:        Running command: XDG_CONFIG_HOME="/Users/blackxfiied/Library/Application Support/heroic/legendaryConfig" /opt/homebrew/bin/legendary list --third-party
(15:07:10) ERROR:   [Backend]:          Aborting not possible. Could not find a matching abort controller for thFgSR5VBurH1zF8QAnJ9u
(15:07:10) INFO:    [Backend]:          Trying to kill steam.exe
(15:07:10) INFO:    [Backend]:          Killed steam.exe
(15:07:10) INFO:    [Backend]:          Checking if wine version exists: GPTK Wine (DX11/DX12 Only) - wine-7.7
(15:07:10) DEBUG:   [Backend]:          Running Wine command: wineboot -k
```

### Steps to reproduce

2. Click on '...'
3. Scroll down to '...'

### Expected behavior

the --wine argument is specified in the launch command

### Screenshots

![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/41133734/f15864f8-a554-47e9-9294-c082c66a9988)

### Heroic Version

Latest Stable

### System Information

macOS 14.0 Beta (23A5328b)
MacBook Pro (14", 2021, M1 Pro, 16/512)

### Additional information

please also look at my other issue that involves 'compatibility_layers.ts'
thanks and sorry to bombard with so many issues

**Comment by Mrredstone5230** [comment]
I can confirm the bug happening on Heroic 2.10.0. To fix this, I was able to force the WINEPREFIX variable as the absolute path to the prefix in the advanced settings of the game. It's annoying, but at least it works...

**Comment by madr7d** [comment]
> I can confirm the bug happening on Heroic 2.10.0. To fix this, I was able to force the WINEPREFIX variable as the absolute path to the prefix in the advanced settings of the game. It's annoying, but at least it works...

Can you tell me how you did that please, I have exactly the same problem :/

**Comment by Mrredstone5230** [comment]
So, in the "Advanced' tab in the settings of a specific game (In this case Satisfactory), you can find the "Environment Variables" section.
<img width="732" alt="image" src="https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/45701824/b015240b-98e5-4669-8bfc-c934d291cb6e">
You can then set the WINEPREFIX variable to the path to your game's prefix (Like on the picture, but you change "user" to your user name). That's what fixed it for me.

**Comment by madr7d** [comment]
Thanks for your answer :) . However, after forcing WINEPREFIX, I launch the game and nothing happens, the play now button reappears, which is annoying.

**Comment by Mrredstone5230** [comment]
Have you made sure the the prefix you're using is in Windows 10? Maybe you could also try making a new prefix, using the command in the GPTK readme.. Oh, and for some reason, I also had issues running directX 11 with gptk. Maybe if you had a program argument that forces use of directX 12 could work?(For instance, that was a problem I had with Satisfactory, which is now on Unreal Engine 5)...

**Comment by madr7d** [comment]
Yes, the prefix I use is in Windows 10. I also tried to create a new prefix but same issue. Force the use of directX ? How ?

**Comment by sanghoonio** [comment]
Even if you force the prefix setting per game, any attempt to edit winecfg or install additional components (UE4) results in everything done in a new default wineprefix. Heroic is not reading in your wineprefix setting at all.

**Comment by Mrredstone5230** [comment]
This might be an UE thing, but I just have to add the -dx12 argument in the game arguments section, although this might just be a Satisfactory thing.

This might also just be an issue with gptk and your game then. The way I was able to confirm that the bug existed is that I realized that my Satisfactory save files were in the default wine prefix directory, and not the one specified in Heroic. The fact that you can't launch your game is, IMO, an unrelated problem with the game prefix, and more of a problem of compatibility with GPTK.

**Comment by eliu9395** [comment]
@Mrredstone5230 Do I have to set this variable for each game I install using gptk? Or is there a way to make gptk games automatically install to the heroic "prefixes" folder?
If not, can I just manually move the files from inside the .wine folder to a new folder?

**Comment by Mrredstone5230** [comment]
I unfortunately don't think there is, sorry.

**Comment by arielj** [comment]
I understand this is fixed now, please try with 2.15.0

feel free to comment if it's still broken and I can reopen this
